runtime/permissions/lib.rs

```diff
@@ -3767,6 +3839,11 @@ pub trait PermissionDescriptorParser: Debug + Send + Sync {
     path: &str,
   ) -> Result<PathQueryDescriptor, PathResolveError>;
 
+  fn parse_net_query(
+    &self,
+    text: &str,
+  ) -> Result<NetDescriptor, NetDescriptorParseError>;
```

Comment:
One tiny thing, is considering this is permissions, I wonder if we should use a different type for query instead of `NetDescriptor` so there's a bit more protection in not accidentally mixing them up? I'm not sure it makes sense. Just a thought.

Reply:
That was my first try. It's hard to walk through it all but I essentially hit a roadblock implementing `QueryDescriptor::from_allow()` (this method fundamentally doesn't make sense) for `NetQuery { type AllowDesc = NetDescriptor }`, and the change exploded in scope trying to address everything. So I restarted.

However, I'm also doubtful we should have separate 'query' and 'allow-listed range' types, since the latter should also be query-able through the runtime API. There might have been technical causes to do this `type AllowDesc` stuff for env or run permissions, but from a wider lens it seems flawed.